ed_robocup/src/gui_plugin.cpp

```diff
@@ -349,16 +313,6 @@ bool RobocupPlugin::srvFitEntityInImage(ed_robocup_msgs::FitEntityInImage::Reque
         ed::WorldModel wm = *world_;
         wm.update(*update_req_);
         ed::EntityConstPtr e = wm.getEntity(entity_id);
-
-        EntityRepresentation2D repr2d = fitter_.GetOrCreateEntity2D(e);
-        if (!repr2d.shape_2d.empty())
-        {
-            geo::Transform2 pose_2d;
-            pose_2d.t = geo::Vec2(fitted_pose.t.x, fitted_pose.t.y);
-            pose_2d.R = geo::Mat2(fitted_pose.R.xx, fitted_pose.R.xy,
-                                  fitted_pose.R.yx, fitted_pose.R.yy);
-            map_filter_.setEntityPose(pose_2d, repr2d.shape_2d, map_filter_padding_);
-        }
```

Comment:
This is the only connection between map updater and fitter.
This part makes sure that the newly fitted entity is added to the (gmapping) map. 
I know that this currently breaks this functionality, but I think there are better ways of implementing this. Right now the map is only updated if the new entity comes from the gui fitter, but this should react to all changes in position of entities within the worldmodel.
One way to do that is to check the revision number of ed perhaps.

Reply:
Why should I accept any changes that breaks stuff. If you already have another solution in mind, please implement that.

Reply:
Hence the draft, I wanted to discuss a potentiap solution first

Reply:
Ok